CMakeLists.txt

```diff
@@ -583,7 +586,7 @@ if (encryption)
 		FIND_PACKAGE(OpenSSL REQUIRED)
 	endif()
 	target_compile_definitions(torrent-rasterbar PUBLIC TORRENT_USE_OPENSSL TORRENT_USE_LIBCRYPTO)
-	target_include_directories(torrent-rasterbar PUBLIC ${OPENSSL_INCLUDE_DIR})
+	target_include_directories(torrent-rasterbar SYSTEM PUBLIC ${OPENSSL_INCLUDE_DIR})
```

Comment:
This is causing gcc to use `-isystem` to specify the include dir which is messing up the include search order and causing build failures for me. I don't think we should be using `SYSTEM` here, nor for the boost include dir below.

Reply:
> …and causing build failures for me.

How that's possible?

Reply:
The exact error is:

> In file included from /usr/x86_64-pc-linux-gnu/include/boost/config/platform/linux.hpp:15:0,
>                  from /usr/x86_64-pc-linux-gnu/include/boost/config.hpp:57,
>                  from /home/steve/projects/libtorrent/include/libtorrent/config.hpp:40,
>                  from /home/steve/projects/libtorrent/src/web_connection_base.cpp:33:
> /usr/x86_64-pc-linux-gnu/include/c++/6.4.0/cstdlib:75:25: fatal error: stdlib.h: No such file or directory
>  #include_next <stdlib.h>

The GNU C/C++ headers are very sensitive to the system include dir search order so it's best not to mess with it.

Reply:
Thanks. So, that is the old #include_next problem... Probably pkg-config for OpenSSL on your system lists /usr/include?

Reply:
Yep. To be exact it's `/usr/x86_64-pc-linux-gnu/include` due to multiarch.

Reply:
So neither shall we put standard include directories in pkg-config (PR updated).

Reply:
> Yep. To be exact it's /usr/x86_64-pc-linux-gnu/include due to multiarch.

Aha! And could you test, please, does CMAKE_C_IMPLICIT_INCLUDE_DIRECTORIES contain that path?

Reply:
I wanted to silence warnings from the boost headers, but if there are still distros with errors in their package configurations, it's not a big deal to revert that.